static/app/utils/useCopyToClipboard.tsx

```diff
@@ -0,0 +1,62 @@
+import {useCallback, useRef, useState} from 'react';
+
+import {addErrorMessage, addSuccessMessage} from 'sentry/actionCreators/indicator';
+import {t} from 'sentry/locale';
+
+type Opts = {
+  text: string;
+  errorMessage?: string;
+  hideMessages?: boolean;
+  onCopy?: undefined | ((copiedText: string) => void);
```

Comment:
Do you need the `undefined |` if it's already optional?

Reply:
yeah we do. so callers can do `<Foo onCopy={undefined} />`

it's handy in places where the prop is optional, like `onCopy={isGood ? fn : undefined}` kind of thing. 

You can play with it using a setup like this: https://www.typescriptlang.org/play?#code/C4TwDgpgBAysCGxoF4oG8BQVtQHbwFsIAuKAZ2ACcBLXAcwG4sd46IB+U3AVwICMIlJjnIQAHqW64AJhABmtCNKgAfKAHIC8ADYR1qjXIhbd64Tm0B7AMaJql3JyhTZC3EoMUa9JgF8MGNYOFFDw0oSkcIgo6MzY+ESk6gCC4QTqADRxohLOMvKK0lm+TEG4IQLAoJEISFComCIJJBoAQhBVIJnZZOJJRiZ6WSJWtsD2uJL5bkrFpcHAULaU2jXR9bFNhC3qAMLwK90ivbkuBe5F2aN2DkkAogfAABbdvkA